### PR TITLE
Add storageType field to HouseTable Entity

### LINKED ISSUE #80: Add storageType field to HouseTable Entity
Now that there is a new storageType field we should add it to the HouseTable entity as well.

### PR DESCRIPTION
**Pull request by ctrezzo**
## Summary

[Issue](https://github.com/linkedin/openhouse/issues/#80) Added a storageType field to the HouseTable Entity to support multiple storage types.

Resolves #80 

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [x] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [x] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

Added storageType field to the HouseTable DTO.

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [x] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

**Review by jainlavina — COMMENTED** [review]
We should add a test to check backwards compatibility. Check that if storageType is not populated in the database then it does not break anything.

File: iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/model/HouseTable.java [hunk 1]
```diff
@@ -39,4 +39,6 @@ public class HouseTable {
   private long lastModifiedTime;
 
   private long creationTime;
+
+  private String storageType;
```
Comment (jainlavina) [inline iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/model/HouseTable.java:43]: We don't have a lot of comments here describing the fields. At least this one deserves some explanation with pointer to the list of supported types.

Comment (jainlavina) [reply iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/model/HouseTable.java:43]: Also, has this column been already added to the schema of all deployed databases?

Comment (HotSushi) [reply iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/model/HouseTable.java:43]: this column has been added to the deployed database schema already

**Comment by ctrezzo** [comment]
Please ignore pull update. Fixing config issues and then will re-update request.

**Comment by ctrezzo** [comment]
Hey @HotSushi! I have updated the pull request. Please take a look. Here are the main changes:

1. storageType field is now populated with the HDFS default
2. I added a unit test to validate defaults are set in the HouseTable builder properly
3. I did a manual test with docker containers where I created a table via a spark shell and verified that the user_table_row table had the new row with "hdfs" set as the value for the storage_type column.
4. All unit tests pass.

File: iceberg/openhouse/internalcatalog/src/test/java/com/linkedin/openhouse/internal/catalog/model/HouseTableTest.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+          Object value = getter.invoke(ht);
+
+          String fieldName =
+              getter.getName().substring(3, 4).toLowerCase() + getter.getName().substring(4);
+          if (!HouseTableSerdeUtils.HTS_FIELD_NAMES.contains(fieldName)) {
+            // not a field getter
+            continue;
+          }
+
+          if (fieldName.equals("storageType")) {
```
Comment (HotSushi) [inline iceberg/openhouse/internalcatalog/src/test/java/com/linkedin/openhouse/internal/catalog/model/HouseTableTest.java:36]: Thanks for this unit test!

**Review by HotSushi — COMMENTED** [review]
Its looking good, can we add a screenshot of the request received in HTS service,  maybe from the docker log?

**Comment by ctrezzo** [comment]
@HotSushi Attached are the screenshots from the docker log during the manual test. Thanks!

Screenshot 2024-05-03 at 2 03 50 PM

Screenshot 2024-05-03 at 2 03 23 PM

Screenshot 2024-05-03 at 2 00 31 PM

**Comment by ctrezzo** [comment]
@HotSushi Attaching screenshot from HTS as well.

Screenshot 2024-05-06 at 12 46 02 PM

**Comment by ctrezzo** [comment]
Taking a look at the failed test

**Review by HotSushi — APPROVED** [review]
LGTM! thanks for the extensive testing in this PR, unit tests and docker screenshots made it very easy to review

**Pull request merged**